src/main/java/org/prebid/server/bidder/loyal/LoyalBidder.java

```diff
@@ -0,0 +1,155 @@
+package org.prebid.server.bidder.loyal;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.appush.proto.AppushImpExtBidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderCall;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.loyal.ExtImpLoyal;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.BidderUtil;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class LoyalBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpLoyal>> LOYAL_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private static final String PUBLISHER_PROPERTY = "publisher";
+    private static final String NETWORK_PROPERTY = "network";
+    private static final String BIDDER_PROPERTY = "bidder";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public LoyalBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<HttpRequest<BidRequest>> httpRequests = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            final ExtImpLoyal extImpLoyal;
+            try {
+                extImpLoyal = parseExtImp(imp);
+            } catch (PreBidException e) {
+                return Result.withError(BidderError.badInput(e.getMessage()));
```

Comment:
this is now incorrect, because it fails on the first parsing error, but we should fail only if all of them are failed